pkg/utils/k8sutil.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"context"
+	"time"
+
+	"github.com/keikoproj/addon-manager/pkg/client/informers/externalversions/internalinterfaces"
+	"k8s.io/client-go/tools/cache"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
+
+func NewWorkflowInformer(dclient dynamic.Interface, ns string, resyncPeriod time.Duration, indexers cache.Indexers, tweakListOptions internalinterfaces.TweakListOptionsFunc) cache.SharedIndexInformer {
+	resource := schema.GroupVersionResource{
+		Group:    "argoproj.io",
+		Version:  "v1alpha1",
+		Resource: "workflows",
+	}
+	informer := NewFilteredUnstructuredInformer(
+		resource,
+		dclient,
+		ns,
+		resyncPeriod,
+		indexers,
+		tweakListOptions,
+	)
+	return informer
+}
+
+func NewFilteredUnstructuredInformer(resource schema.GroupVersionResource, client dynamic.Interface, namespace string, resyncPeriod time.Duration, indexers cache.Indexers, tweakListOptions internalinterfaces.TweakListOptionsFunc) cache.SharedIndexInformer {
+	ctx := context.Background()
+	return cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				if tweakListOptions != nil {
+					tweakListOptions(&options)
+				}
+				return client.Resource(resource).Namespace(namespace).List(ctx, options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				if tweakListOptions != nil {
+					tweakListOptions(&options)
+				}
+				return client.Resource(resource).Namespace(namespace).Watch(ctx, options)
+			},
+		},
+		&unstructured.Unstructured{},
+		resyncPeriod,
+		indexers,
+	)
+}
+
+func TweakListOptions(options *metav1.ListOptions) {
```

Comment:
Why is this empty function needed?

Reply:
For generating workflowinformer

Reply:
Let's remove empty functions, not useful.

Reply:
f70a7281b87b34b6999fbdcae5db6ccd5f443613